:sparkles: block move with annotation (#8690)
kubernetes-sigs/cluster-api: docs/book/src/reference/labels_and_annotations.md

| clusterctl.cluster.x-k8s.io/delete-for-move                      | DeleteForMoveAnnotation will be set to objects that are going to be deleted from the source cluster after being moved to the target cluster during the clusterctl move operation. It will help any validation webhook to take decision based on it.                                                                                                                                                                                                                                                                                                         |

> | clusterctl.cluster.x-k8s.io/block-move                           | BlockMoveAnnotation if defined on at least one of the objects in scope for a cluster move operation, prevents the operation to start. Providers' controllers are expected to set the annotation on resources that cannot be instantaneously paused, and remove the annotation when the resource has been actually paused.                                                                                                                                                                                                                                   |

Reviewer: ```suggestion
| clusterctl.cluster.x-k8s.io/block-move                           | BlockMoveAnnotation prevents the cluster move operation from starting if it is defined on at least one of the objects in scope. Provider controllers are expected to set the annotation on resources that cannot be instantaneously paused and remove the annotation when the resource has been actually paused.                                                                                            |
```
^ Update doc to match the doc comment in the type. I think you'd want to reformat the number of spaces between the `|`, though.

Author: Updated
